benchmarks/build.gradle

```diff
@@ -58,55 +66,65 @@ task jmhProfilers(type: JavaExec, description:'Lists the available profilers for
-  }
+  @TaskAction
+  void exec() {
+    main = 'org.openjdk.jmh.Main'
 
-  def exclude = project.properties.get('exclude');
-  def format = project.properties.getOrDefault('format', 'json');
-  def profilers = project.properties.get('profilers');
-  def jvmArgs = project.properties.get('jvmArgs')
-  def verify =  project.properties.get('verify');
+    def include = ''
+    if (project.hasProperty('include')) {
+      include = [".*" + project.properties.get('include') + ".*"]
+    }
+    if (project.hasProperty('fullInclude')) {
+      include = project.properties.get('fullInclude');
+    }
 
-  def resultFile = file("build/reports/jmh/result.${format}")
+    def exclude = project.properties.get('exclude');
+    def format = project.properties.getOrDefault('format', 'json');
+    def profilers = project.properties.get('profilers');
+    def jvmArgs = project.properties.get('jvmArgs')
+    def verify =  project.properties.get('verify');
 
-  args include
-  if(exclude) {
-    args '-e', exclude
-  }
-  if(verify != null) { // execute benchmarks with the minimum amount of execution (only to check if they are working)
-    println "≥≥ Running in verify mode"
-    args '-f' , 1
-    args '-wi' , 1
-    args '-i' , 1
-  }
-  args '-foe', 'true'   //fail-on-error
-  args '-v', 'NORMAL'   //verbosity [SILENT, NORMAL, EXTRA]
-  if(profilers) {
-    profilers.split(',').each {
-      args '-prof', it
+    def resultFile = project.file("build/reports/${this.name}/result.${format}")
+
+    args include
+    if(exclude) {
+      args '-e', exclude
     }
-  }
-  args '-jvmArgsPrepend', '-Xmx3072m'
-  args '-jvmArgsPrepend', '-Xms3072m'
-  if(jvmArgs) {
-    for(jvmArg in jvmArgs.split(' ')) {
-      args '-jvmArgsPrepend', jvmArg
+    if(verify != null) { // execute benchmarks with the minimum amount of execution (only to check if they are working)
+      println "≥≥ Running in verify mode"
+      args '-f' , 1
+      args '-wi' , 1
+      args '-i' , 1
     }
-  }
-  args '-rf', format
-  args '-rff', resultFile
+    args '-foe', 'true'   //fail-on-error
+    args '-v', 'NORMAL'   //verbosity [SILENT, NORMAL, EXTRA]
+    if(profilers) {
+      profilers.split(',').each {
+        args '-prof', it
+      }
+    }
+    args '-jvmArgsPrepend', '-Xmx3072m'
+    args '-jvmArgsPrepend', '-Xms3072m'
+    if(jvmArgs) {
+      for(jvmArg in jvmArgs.split(' ')) {
+        args '-jvmArgsPrepend', jvmArg
+      }
+    }
+    args '-rf', format
+    args '-rff', resultFile
 
-  doFirst {
     println "\nExecuting JMH with: $args \n"
     resultFile.parentFile.mkdirs()
+    super.exec()
   }
-}
\ No newline at end of file
+}
+
+task jmh(type: JmhExec, description: 'Executing JMH benchmarks') {
+  classpath = sourceSets.main.runtimeClasspath + configurations.current
+}
+
+task jmhBaseline(type: JmhExec, description: 'Executing JMH baseline benchmarks') {
+  classpath = sourceSets.main.runtimeClasspath + configurations.baseline
+}
```

Comment:
suggestion: add a task that runs benchmark(s) on the current then on the baseline:

- most parameters can be shared except the `resultFile` one :+1:
- most probably, the two jmh runs cannot be parallelized due to how the jar gets generated

Reply:
I was running both with:
```
$ ../gradlew jmhBaseline jmh -Pinclude=ShakespearePlaysScrabbleRegular -Pformat=text
```

works like a charm :)

Reply:
sounds good enough. maybe rename `jmh` to `jmhCurrent`?

Reply:
I thought about it, but decided to keep it "as is" because everyone got used to `jmh` (as the default for the "current" benchmarks)... WDYT?

Reply:
yeah, not a strong opinion on said naming. actually your argument that it is the easiest to naturally discover is sound, let's keep `jmh` then 👍